frontend/src/modules/WellCompletion/wellCompletionsDataAccessor.ts

```diff
@@ -0,0 +1,236 @@
+ */
+export const TimeAggregations = {
+    None: (arr: number[]): number => arr[arr.length - 1],
+    Max: (arr: number[]): number => Math.max(...arr),
+    Average: (arr: number[]): number => arr.reduce((a, b) => a + b) / arr.length,
+};
+export type TimeAggregation = keyof typeof TimeAggregations;
+
+export class WellCompletionsDataAccessor {
+    /**
+     * This class is responsible for parsing and providing data for the well completions plot.
+     *
+     * It parses data structure received from backend and creates the data structure needed for the plot
+     * in WellCompletionsPlot-component.
+     *
+     * The data is parsed from the WellCompletionData_api object and methods are taken from the
+     * WellCompletions-component of webviz-subsurface-components package, and modified to fit.
+     */
+    private _dataSet: WellCompletionDataSet_api | undefined;
+    private _subzones: WellCompletionZone_api[];
+    private _wells: WellCompletionWell_api[];
+    private _searchWellText: string;
+    private _hideZeroCompletions: boolean;
+
+    constructor() {
+        this._dataSet = undefined;
+        this._subzones = [];
+        this._wells = [];
+        this._searchWellText = "";
+        this._hideZeroCompletions = false;
+    }
+
+    parseWellCompletionsData(data: WellCompletionData_api): void {
+        // TODO:
+        // - Filter wells when filter functionality is in place
+        // - Filter subzones when filter functionality is in place
+
+        this._dataSet = data.json_data;
+        this._wells = this._dataSet.wells;
+
+        // Exctract all subzones
+        this._subzones = [];
+        this._dataSet.stratigraphy.forEach((zone) => this.findSubzones(zone, this._subzones));
+    }
+
+    setNumberOfWellsToShow(numberOfWellsToShow: number): void {
+        // TODO: Add functionality for pagination or similar to control number of wells to include in plotData.
+        void numberOfWellsToShow;
+        return;
+    }
+
+    setSearchWellText(searchWell: string): void {
+        this._searchWellText = searchWell;
+    }
+
+    setHideZeroCompletions(hideZeroCompletions: boolean): void {
+        this._hideZeroCompletions = hideZeroCompletions;
+    }
+
+    getTimeSteps(): string[] {
+        if (this._dataSet === undefined) return [];
+        return this._dataSet.timeSteps;
+    }
+
+    createPlotData(timeStepSelection: string | [string, string], timeAggregation: TimeAggregation): PlotData | null {
+        // TODO: Consider removing function arguments, and use setter-methods for each argument and set to an attribute.
+        //       This would make it easier to modify/adjust single attributes and call "createPlotData" again.
+        if (!this._dataSet) return null;
+
+        let timeStepIndexRange: [number, number] | null = null;
+        if (typeof timeStepSelection === "string") {
+            const timeStepIndex = this._dataSet ? this._dataSet.timeSteps.indexOf(timeStepSelection) : 0;
```

Comment:
Consider `const timeStepIndex = this._dataSet?.timeSteps.indexOf(timeStepSelection) ?? 0;`

Reply:
Replace with only time step index. If -1 is returned the new if-check  should return null. If time step selections are not valid, no incorrect data should be shown